src/main/java/at/hannibal2/skyhanni/data/mob/Mob.kt

```diff
@@ -166,42 +178,44 @@ class Mob(
         fullEntityList().associateWith { this }
 
     internal fun internalAddEntity(entity: EntityLivingBase) {
+        internalRemoveColor()
         if (baseEntity.entityId > entity.entityId) {
             extraEntitiesList.add(0, baseEntity)
-            internalRemoveColor()
             baseEntity = entity
-            internalHighlight()
         } else {
             extraEntitiesList.add(extraEntitiesList.lastIndex + 1, entity)
         }
+        internalHighlight()
         updateBoundingBox()
         MobData.entityToMob[entity] = this
     }
 
     internal fun internalAddEntity(entities: Collection<EntityLivingBase>) {
         val list = entities.drop(1).toMutableList().apply { add(baseEntity) }
-        extraEntitiesList.addAll(0, list)
         internalRemoveColor()
+        extraEntitiesList.addAll(0, list)
         baseEntity = entities.first()
         internalHighlight()
         updateBoundingBox()
         removeExtraEntitiesFromChecking()
         MobData.entityToMob.putAll(entities.associateWith { this })
     }
 
-    internal fun internalUpdateOfEntity(entity: EntityLivingBase) = when (entity.entityId) {
-        baseEntity.entityId -> {
-            internalRemoveColor()
-            baseEntity = entity
-            internalHighlight()
-        }
-
-        armorStand?.entityId ?: Int.MIN_VALUE -> armorStand = entity as EntityArmorStand
-        else -> {
-            extraEntitiesList.remove(entity)
-            extraEntitiesList.add(entity)
-            Unit // To make return type of this branch Unit
+    internal fun internalUpdateOfEntity(entity: EntityLivingBase) {
+        internalRemoveColor()
+        when (entity.entityId) {
```

Comment:
i dont understand this line

Reply:
I don't understand what you don't understand. Please elaborate.
Since this is a simple when to switch between which entity is for the update, eg basEntity, Armorstand or else it must be in the extra list.
The only thing I did new here was that I remove the highlight for the old entity and activated it again for the new ones.
And for the question why we need it: Hypixel sends sometimes 2 spawn packets for one and the same entity which is handheld from minecraft that the first one is invalid and "removes" the first entity and then spawns a new one. The issue is we pull so fast that we can get this first "wrong" entity, which doesn't matter in the values since the are literally the same but the reference gets brocken and therefore it won't  update the game state for the entity any longer.